realm/realm-library/src/main/java/io/realm/sync/permissions/Permission.java

```diff
@@ -0,0 +1,562 @@
+            canRead = false;
+            canUpdate = false;
+            canDelete = false;
+            canSetPermissions = false;
+            canQuery = false;
+            canCreate = false;
+            canModifySchema = false;
+            return this;
+        }
+
+        /**
+         * Disables all privileges.
+         */
+        public Builder noPrivileges() {
+            canRead = true;
+            canUpdate = true;
+            canDelete = true;
+            canSetPermissions = true;
+            canQuery = true;
+            canCreate = true;
+            canModifySchema = true;
+            return this;
+        }
+
+        /**
+         * Defines if this role can read from given resource or not.
+         *
+         * <ol>
+         *     <li>
+         *         <b>Realm:</b>
+         *         The role is allowed to read all objects from the Realm. If {@code false}, the
+         *         Realm will appear completely empty to the role, effectively making it inaccessible.
+         *     </li>
+         *     <li>
+         *         <b>Class:</b>
+         *         The role is allowed to read the objects of this type and all referenced objects.
+         *         If {@code false}, the role cannot see any object of this type and all queries
+         *         against the type will return no results.
+         *     </li>
+         *     <li>
+         *         <b>Object:</b>
+         *          Determines if a role is allowed to see the individual object or not.
+         *     </li>
+         * </ol>
+         *
+         * @param canRead {@code true} if the role is allowed to read this resource, {@code false} if not.
+         */
+        public Builder canRead(boolean canRead) {
+            this.canRead = canRead;
+            return this;
+        }
+
+        /**
+         * Defines if this role can update the given resource or not.
+         *
+         * <ol>
+         *     <li>
+         *         <b>Realm:</b>
+         *         Not applicable.
+         *     </li>
+         *     <li>
+         *         <b>Class:</b>
+         *         Not applicable.
+         *     </li>
+         *     <li>
+         *         <b>Object:</b>
+         *         If {@code true}, the role is allowed to update properties on the object, but not
+         *         set or change permissions.
+         *     </li>
+         * </ol>
+         *
+         * @param canUpdate {@code true} if the role is allowed to update this resource, {@code false} if not.
+         */
+        public Builder canUpdate(boolean canUpdate) {
+            this.canUpdate = canUpdate;
+            return this;
+        }
+
+        /**
+         * Defines if this role can delete the given resource or not.
+         *
+         * <ol>
+         *     <li>
+         *         <b>Realm:</b>
+         *         Not applicable.
+         *     </li>
+         *     <li>
+         *         <b>Class:</b>
+         *         Not applicable.
```

Comment:
Theoretically applicable, since it determines whether the user can remove the class from the Realm. But in practice not applicable because destructive schema changes are disallowed.

Reply:
I'll rather leave it as N/A for now. No reason to complicate things with possible future enhancements.